src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1856,6 +1856,23 @@ void Arguments::set_heap_size() {
       reasonable_max = MAX2(reasonable_max, (julong)MinHeapSize);
     }
 
+    // With current TSan memory mapping of 48bit aarch64, it's very hard to allocate large continuous space.
+    // Add a limitation to avoid the allocation failure in machine with huge physical RAM.
+    // Does not support 39bit and 42bit aarch64 right now. Will add it if necessary.
+    TSAN_ONLY(
+#if defined(AARCH64)
+#define MAX_AVAILABLE_VM_SPACE_IN_AARCH64 (1024ULL * 1024ULL * 1024ULL * 16ULL) //16GB
```

Comment:
The macros could be moved to tsan.hpp, and prefixed with "TSAN_", e.g. "TSAN_MSB_IN_AARCH64".
Or we could just use C++ const variable inline.

"\_\_FUNCTION\_\_" could also be used directly inline.

Reply:
I just see other functions have similar style, which define and use the Macros or const variables locally, due to merely one use place.

Reply:
Those are likely from ancient code. Upstream OpenJDK is also moving away from that style.
See
https://wiki.openjdk.java.net/display/HotSpot/StyleGuide (search "Macros").